tests/cxx/elaborated_type.cc

```diff
@@ -60,6 +60,21 @@ void global_qualified_elab(class ::GlobalClass* c);
 // they must also be forward-declared.
 Elaboration::Template<int, float>* namespace_qualified_template;
 
+// Elaborated types can be treated as forward-declarations, hence no fwd-decl is
+// needed if a declaration is introduced by elaborated type.
+// A declaration "owned" by elaborated type is an implicit declaration
+// introduced by elaborated type if no one was already present.
+void owning_elaborated(class OwningElaborated*);
+
+class OwningElaborated;
```

Comment:
I'd also be interested in seeing a use of `OwningElaborated` without elaboration and before the first real forward-declare, e.g.
```
void f(class OwningElaborated*);
void g(OwningElaborated*);
class OwningElaborated;
void h(OwningElaborated*);
```

That would make it clearer that the first elaborated type specifier really pushes a decl out into the containing scope.